backend/src/test/java/com/google/backend/servlets/AlertsDataServletTest.java

```diff
@@ -0,0 +1,79 @@
+package com.google.backend.servlets;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.models.Alert;
+import com.google.models.Anomaly;
+import com.google.models.Timestamp;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito.*;
+import org.mockito.MockitoAnnotations;
+
+/** Contain tests for methods in {@link AlertsDataServlet} class. */
+@RunWith(JUnit4.class)
+public class AlertsDataServletTest {
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+
+  private static final String CONTENT_TYPE = "application/json;";
+  private static final String ENTITY_KIND = Alert.ALERT_ENTITY_KIND;
+
+  private static final AlertsDataServlet alertsDataServlet = new AlertsDataServlet();
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(
+    new LocalDatastoreServiceTestConfig());
+  private StringWriter stringWriter = new StringWriter();
```

Comment:
if you're creating the stringWriter as the object on a class level, what happens when you have multiple @Test that all access this same `stringWriter`, instead of a per test `stringWriter`. Would it make a difference?

Reply:
In the `setUp` function a new `PrintWriter` is being created for each test so it's like having a different `stringWriter` for each test. If there was only one `stringWriter` for all the tests it wouldn't work as the `stringWriter` needs to be "cleared" (otherwise the previous responses would still be in the `stringWriter`). Does that make sense?

Reply:
I see, so because you're doing the new PrintWriter(stringWriter), there will abe a unique stringWriter.

OK, thats ok then, just wanted to make sure the stringWriter will not clash when there are new tests.